assets/opengrep_rules/client/service-worker-origin-validation.yaml

```diff
@@ -0,0 +1,39 @@
+rules:
+  - id: chromium-service-worker-origin-validation
+    metadata:
+      author: Andrea Brancaleoni <abc@pompel.me>
+      references:
+        - https://chromium.googlesource.com/chromium/src/+/main/docs/security/service-worker-security-faq.md
+        - https://github.com/brave/brave-browser/wiki/Security-reviews
+      source: https://github.com/brave/security-action/blob/main/assets/opengrep_rules/client/service-worker-origin-validation.yaml
+      assignees: |
+        thypon
+        diracdeltas
+        bridiver
+      category: security
+    languages: [cpp, c]
+    message: |
+      Service Worker operations via ServiceWorkerContext must validate origins to prevent
```

Comment:
I'm pretty sure this rule is not necessary as service workers internally enforce same origin restrictions.